gprofiler/utils.py

```diff
@@ -97,12 +104,28 @@ def start_process(cmd: Union[str, List[str]], via_staticx: bool, **kwargs) -> Po
             env = env if env is not None else os.environ.copy()
             env.update({"LD_LIBRARY_PATH": ""})
 
+    cur_preexec_fn = kwargs.pop("preexec_fn", os.setpgrp)
+
+    if term_on_parent_death:
+        def wrap_with_set_pseg_death(inner):
+            PR_SET_PDEATHSIG = 1
+            def wrapper():
+                ret = prctl(PR_SET_PDEATHSIG, signal.SIGTERM)
+                if ret != 0:
+                    logger.warning("set_pdeathg on child process failed")
```

Comment:
Log errno here

Reply:
There's a way to make `ctypes` track the errno & give it to you

Reply:
done